src/lib.rs

```diff
@@ -517,176 +531,203 @@ impl State {
             .texture
             .create_view(&wgpu::TextureViewDescriptor::default());
 
-        self.context.begin_frame(self.egui_state.take_egui_input(window));
+        self.context
+            .begin_frame(self.egui_state.take_egui_input(window));
 
-        egui::Window::new(env!("CARGO_PKG_NAME"))
-            .title_bar(true)
-            .show(&self.context, |ui| {
-                egui::trace!(ui);
+        #[cfg(target_arch = "wasm32")]
+        let web_fullscreen = web_sys::window()
+            .and_then(|win| win.screen().ok().zip(Some(win)))
+            .map(|(screen, win)| {
+                screen.width().ok()
+                    == win
+                        .inner_width()
+                        .ok()
+                        .and_then(|v| v.as_f64())
+                        .map(|v| v as i32)
+                    && screen.height().ok()
+                        == win
+                            .inner_height()
+                            .ok()
+                            .and_then(|v| v.as_f64())
+                            .map(|v| v as i32)
+            })
+            .unwrap_or(false);
 
-                ui.label(format!(
-                    "Version {} ({}{}{})",
-                    env!("CARGO_PKG_VERSION"),
-                    std::env::consts::OS,
-                    if std::env::consts::OS.is_empty() {
-                        ""
-                    } else {
-                        " "
-                    },
-                    std::env::consts::ARCH
-                ));
-                ui.label(format!("Render backend: {}", self.backend));
-                ui.label(format!(
-                    "Last frame: {:.1}ms ({:.0} FPS)",
-                    self.prev_frame_time.as_micros() as f64 / 1000.0,
-                    1.0 / self.prev_frame_time.as_secs_f64()
-                ));
-                #[cfg(not(target_arch = "wasm32"))]
-                ui.label("Fullscreen: [F11]");
-                ui.separator();
+        #[cfg(not(target_arch = "wasm32"))]
+        let web_fullscreen = false;
 
-                let settings_clone = self.settings.clone();
+        if !(window.fullscreen().is_some() || web_fullscreen) {
```

Comment:
I think disabling the settings window should be separate from fullscreen, since people may want to change the settings whilst in fullscreen

Reply:
Could use a keybind I suppose